server/api/lib/mavat/index.js

```diff
@@ -39,22 +39,99 @@ const init = () =>
 		})();
 	});
 
-const downloadFile = (url, file, entityDocId, entityDocNumber) => {
+const downloadChallengedFile = (url, file, options) => {
 	return new Promise((resolve) => {
-		http.get(url, (response) => {
-			response.pipe(file);
-			file.on('finish', async function () {
-				await file.close();
-				resolve(true);
-			});
+		options = options || {};
+
+		http.get(url, options, (response) => {
+			if (response.statusCode !== 200) {
+				Log.info(`downloadChallengedFile failed with status ${response.statusCode} for url ${url}`);
```

Comment:
Log error instead of info? Also, doesn't http.get goes to error if the request fails?

Reply:
no, the error event is triggered when there's a connection error, premature connection close etc.